pkg/testcase/secretsencrypt.go

```diff
@@ -0,0 +1,137 @@
+
+	stdOutput, err := shared.SecretEncryptOps(action, cpIp, product)
+	Expect(err).NotTo(HaveOccurred(), "error: secret-encryption: "+action)
+	verifyStdOut(action, stdOutput)
+	if (action == "reencrypt") || (action == "rotate-keys") {
+		time.Sleep(20 * time.Second)
+	}
+
+	for _, node := range ips {
+		var nodearr []string
+		nodearr = append(nodearr, node)
+		nodeIp, errRestart := shared.ManageService(product, "restart", "server", nodearr)
+		Expect(errRestart).NotTo(HaveOccurred(), "error restart service for node: "+nodeIp)
+		// Order of reboot matters. Etcd first then control plane nodes.
+		// Little lag needed between node restarts to avoid issues.
+		shared.LogLevel("INFO", "Sleep 10 seconds - wait before restarting next node in cluster")
+		time.Sleep(10 * time.Second)
+	}
+	switch product {
+	case "k3s":
+		shared.LogLevel("INFO", "Sleep 30 seconds - wait for services to come up")
+		time.Sleep(30 * time.Second)
+	case "rke2":
+		shared.LogLevel("INFO", "Sleep 60 seconds - wait for services to come up")
+		time.Sleep(60 * time.Second)
+	}
+
+	stdStatusOut, errStatus := shared.SecretEncryptOps("status", cpIp, product)
+	Expect(errStatus).NotTo(HaveOccurred(), "error getting secret-encryption status")
+	verifyStatusOutput(action, stdStatusOut)
+
+	errLog := logEncryptionFileContents(ips, product)
+	Expect(errLog).NotTo(HaveOccurred())
+}
+
+func verifyStdOut(action, stdout string) {
+	switch action {
+	case "prepare":
+		Expect(stdout).To(ContainSubstring("prepare completed successfully"))
+	case "rotate":
+		Expect(stdout).To(ContainSubstring("rotate completed successfully"))
+	case "reencrypt":
+		Expect(stdout).To(ContainSubstring("reencryption started"))
+	case "rotate-keys":
+		Expect(stdout).To(ContainSubstring("keys rotated, reencryption started"))
+	}
+}
+
+func verifyStatusOutput(action, stdout string) {
+	switch action {
+	case "prepare":
+		Expect(stdout).To(ContainSubstring("Encryption Status: Enabled"))
+		Expect(stdout).To(ContainSubstring("Current Rotation Stage: prepare"))
+		Expect(stdout).To(ContainSubstring("Server Encryption Hashes: All hashes match"))
+	case "rotate":
+		Expect(stdout).To(ContainSubstring("Encryption Status: Enabled"))
+		Expect(stdout).To(ContainSubstring("Current Rotation Stage: rotate"))
+		Expect(stdout).To(ContainSubstring("Server Encryption Hashes: All hashes match"))
+	case "reencrypt":
+		Expect(stdout).To(ContainSubstring("Encryption Status: Enabled"))
+		Expect(stdout).To(ContainSubstring("Current Rotation Stage: reencrypt_finished"))
+		Expect(stdout).To(ContainSubstring("Server Encryption Hashes: All hashes match"))
+	case "rotate-keys":
+		Expect(stdout).To(ContainSubstring("Encryption Status: Enabled"))
+		Expect(stdout).To(ContainSubstring("Current Rotation Stage: reencrypt_finished"))
+		Expect(stdout).To(ContainSubstring("Server Encryption Hashes: All hashes match"))
+	}
+}
+
+func getNodeIps(etcdNodes, cpNodes []shared.Node) []string {
+	var sarray []string
```

Comment:
sorry lol, maybe missing something 

what is sarray ?  shared array ?

Reply:
i was thinkin string array in my mind. i could probably rename if to nodeIpsArr or some such. :)